### PR TITLE
Import refactor

### PR DESCRIPTION
**Pull request by sloria**
Implements the package structure described in #5 
- Base classes in `hyp.base`
- `hyp.marshmallow` and `hyp.schematics` packages for integration with 3rd-party libs
- Removes `hyp.adapters` and `adapter_for`
- Bumps version in setup.py -> 0.6.0
- As per the PyPA's recommendation, use setuptools over distutils: https://packaging.python.org/en/latest/current.html#packaging-tool-recommendations

### CODE REVIEW

File: hyp/base.py [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+class BaseResponder(object):
     TYPE = None
     SERIALIZER = None
     LINKS = None
+    ADAPTER = None
 
     def __init__(self):
-        self.adapter = adapter_for(self.SERIALIZER)(self.SERIALIZER)
+        if not self.ADAPTER:
+            raise NotImplementedError('Responder must define ADAPTER class variable')
```
Comment (kalasjocke) [inline hyp/base.py:16]: :thumbsup:

**Comment by kalasjocke** [comment]
Thanks for this pull request @sloria. Everything looks great. Since the changes in this pull request isn't backwards compatible I think that we should release a new major version. What's your take on that?

**Comment by sloria** [comment]
@kalasjocke Are you referring to a 1.0 release?

According to [Semver 2.0](http://semver.org/) , breaking changes are expected in pre-1.0 releases:

> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

1.0 denotes that the public API is stable.

Given that this is the first release with this new package structure, I'd recommend incrementing the minor version (0.6.0) for now until you are confident that this is the API you want.

However, I'll let you make the final call on this.

**Comment by kalasjocke** [comment]
@sloria I was thinking of a major release soon anyway since this is such a small library. But perhaps we can merge this change in and see how we feel about it over time and defer the decision.

Again, thanks for implementing this.

**Comment by kalasjocke** [comment]
https://pypi.python.org/pypi/hyp/0.6.0

**Pull request merged**